cpp/include/raft/common/nvtx.hpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2019-2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "detail/nvtx.hpp"
+
+namespace raft {
+namespace common {
+
+/**
+ * @brief Push a named nvtx range
+ * @param format range name format (accepts printf-style arguments)
+ * @param args the arguments for the printf-style formatting
+ */
+template <typename... Args>
+inline void PUSH_NVTX_RANGE(const char* format, Args... args)
+{
+  detail::pushRange(format, args...);
+}
+
+/**
+ * @brief Synchronize CUDA stream and push a named nvtx range
+ * @param format range name format (accepts printf-style arguments)
+ * @param args the arguments for the printf-style formatting
+ * @param stream stream to synchronize
+ */
+template <typename... Args>
+inline void PUSH_NVTX_RANGE(rmm::cuda_stream_view stream, const char* format, Args... args)
```

Comment:
Nitpick: I'd prefer to keep caps for constants and macros. Any reason we can't/shouldn't use lower-case for these?

Reply:
It was in caps in cuml. I suppose, there was some logic or legacy behind this. Maybe to make it stand out of everything else to have a better chance of not forgetting to match push/pop directives?.. Also here in raft the caps may scare off developers and compel them to use the safer `NvtxRange` version :)

Though, I don't mind changing it to the camel case if you prefer that.

Reply:
Thanks. I actually prefer snake case here and I believe there was a general consensus to use it across raft (Thejaswi had created a PR at once point to enforce this but it hasn't yet been merged).

Reply:
That said, there are enough places in raft that are still using camel case but my preference is to start using snake case where possible in the public bits until we do our batch rename (after we establish the public API)

Reply:
Wow, I didn't really know that! sure, I'll change that to snake